File: leetcode/443-StringCompression/solution.go
func compress(chars []byte) int {
	if len(chars) <= 0 {
		return len(chars)
	}
	var prev = chars[0]
	var last int
	var count = 1
	for i:=1; i<len(chars); i++ {
		if chars[i] != prev {
			chars[last] = prev
			last++
			if count > 1 {
				ns := []byte(strconv.Itoa(count))
				for _, d := range ns {
					chars[last] = d
					last++
				}
			}
			prev = chars[i]
			count=1
		} else {
			count++
		}
	}
	chars[last] = chars[len(chars)-1]
	last++
	if count > 1 {
		ns := []byte(strconv.Itoa(count))
		for _, d := range ns {
			chars[last] = d
			last++
		}
	}
	return last
}

File: leetcode/443-StringCompression/Solution.rs
impl Solution {
    pub fn compress(chars: &mut Vec<char>) -> i32 {
        fn fill(chars: &mut Vec<char>, mut pos: usize, ch: char, count: usize) -> usize {
            chars[pos] = ch;
            pos += 1;
            if count != 1 {
                let num = count.to_string();
                for ch in num.chars() {
                    chars[pos] = ch;
                    pos += 1;
                }
            }
            pos
        }
        let mut pos: usize = 0;
        let mut ch: char = chars[0];
        let mut count = 1;
        for i in 1..chars.len() {
            if chars[i] == ch {
                count += 1;
            } else {
                pos = fill(chars, pos, ch, count);
                ch = chars[i];
                count = 1;
            }
        }
        fill(chars, pos, ch, count) as i32
    }
}
